### PR TITLE
[ONNX] Add support for operator `add` between tensor list

### PR DESCRIPTION
**Pull request by BowenBao**
E.g.
```python
outs = []
outs += [torch.randn(3,4)]
outs = outs + [torch.randn(4,5), torch.randn(5,6)]
```

### CODE REVIEW

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 1]
```diff
@@ -2516,8 +2516,8 @@ def forward(self, x):
                 arr = x.split([3, 4, 1, 1, 2, 3, 2], 0)
                 res2 = torch.zeros(3, 4, dtype=torch.long)
                 for i in range(len(arr)):
-                    res = res.append(arr[i].sum(0, False))
-                    res1 = res1.append(arr[-1 - i].sum(0, False))
+                    res = res + [arr[i].sum(0, False)]
```
Comment (KsenijaS) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:2519]: is the add between tensor list sort of a workaround for the issue of of empty list that is populated inside the loop via append?

Comment (BowenBao) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:2519]: The purpose was not to workaround, this happens to be another way of working with list of tensors aside from append, and has to be supported as well. The issue with append will be fixed in another PR.

File: torch/onnx/symbolic_opset9.py [hunk 2]
```diff
@@ -83,6 +83,9 @@ def reshape_as(g, self, other):
 
 
 def add(g, self, other, alpha=None):
+    if isinstance(self.type(), torch._C.Type) and self.type().isSubtypeOf(torch._C.ListType.ofTensors()):
```
Comment (neginraoof) [inline torch/onnx/symbolic_opset9.py:86]: Try _is_value and _is_tensor_list from symbolic helper.

File: torch/onnx/symbolic_opset11.py [hunk 3]
```diff
@@ -279,6 +279,19 @@ def append(g, self, tensor):
     return g.op("SequenceInsert", self, tensor)
 
 
+def add(g, self, other, alpha=None):
+    if isinstance(self.type(), torch._C.Type) and self.type().isSubtypeOf(torch._C.ListType.ofTensors()):
```
Comment (neginraoof) [inline torch/onnx/symbolic_opset11.py:283]: Try _is_value and _is_tensor_list from symbolic helper.

File: test/onnx/test_pytorch_onnx_onnxruntime.py (hunk 1 above)
Comment (neginraoof) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:2519]: Then why are we replacing the tests here? Can you add the new tests without removing old ones?

File: torch/onnx/symbolic_opset11.py (hunk 3 above)
Comment (BowenBao) [reply torch/onnx/symbolic_opset11.py]: Thanks!

File: test/onnx/test_pytorch_onnx_onnxruntime.py (hunk 1 above)
Comment (BowenBao) [reply test/onnx/test_pytorch_onnx_onnxruntime.py]: Good point, restored prev tests and added new one.

**Review by neginraoof — APPROVED** [review]
LGTM. Thanks! Maybe rebase to resolve CI failures.

**Pull request closed without merging**